src/python/test/t0001-pybinding-basic.t

```diff
@@ -0,0 +1,132 @@
+    sanity_check ./logdir/perfflow.$(hostname).[0-9]*.pfw &&
+    rm ./logdir/perfflow.$(hostname).[0-9]*.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: name can be included in filename' '
+    PERFFLOW_OPTIONS="name=mycomponent:log-filename-include=name" \
+	../smoketest.py &&
+    test -f perfflow.mycomponent.pfw &&
+    sanity_check perfflow.mycomponent.pfw &&
+    rm perfflow.mycomponent.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: instance-path included in filename' '
+    PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    test -f perfflow.{[a-f0-9]*}.pfw &&
+    sanity_check perfflow.{[a-f0-9]*}.pfw &&
+    rm perfflow.{[a-f0-9]*}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: filename includes all' '
+    PERFFLOW_OPTIONS="log-filename-include=name,instance-path,hostname,pid" \
+	../smoketest.py &&
+    test -f perfflow.generic.{[a-f0-9]*}.$(hostname).[0-9]*.pfw &&
+    sanity_check perfflow.generic.{[a-f0-9]*}.$(hostname).[0-9]*.pfw &&
+    rm perfflow.generic.{[a-f0-9]*}.$(hostname).[0-9]*.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: filename includes all in correct order' '
+    PERFFLOW_OPTIONS="log-filename-include=hostname,instance-path,pid,name" \
+	../smoketest.py &&
+    test -f perfflow.$(hostname).{[a-f0-9]*}.[0-9]*.generic.pfw &&
+    sanity_check perfflow.$(hostname).{[a-f0-9]*}.[0-9]*.generic.pfw &&
+    rm perfflow.$(hostname).{[a-f0-9]*}.[0-9]*.generic.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: SLURM supported' '
+    SLURM_JOB_ID=123456 SLURM_STEP_ID=1 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "123456.1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: LSF supported' '
+    LSB_JOBID=123456 LS_JOBPID=1 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "123456.1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: SLURM_JOB_ID + STEP_ID must be given' '
+    SLURM_JOB_ID=123456 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: LSB_JOBID along will not work' '
+    LSB_JOB_ID=123456 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: Flux supported' '
+    FLUX_JOB_ID=123456 \
```

Comment:
Looks like flux uses the unicode `ƒ` in the env var.  Might be worth testing that that oddity works here:

```
ƒ(s=1,d=0) fluxuser@dc9ae6ac0bde:~$ flux mini run printenv | grep FLUX_JOB_ID
FLUX_JOB_ID=ƒeF9QZG3
```

Reply:
Thanks.